packages/lesswrong/components/ea-forum/TargetedJobAdSection.tsx

```diff
@@ -7,149 +7,159 @@ import { useCurrentUser } from '../common/withUser';
+  const [activeJob, setActiveJob] = useState<string>()
   
   // select a job ad to show to the current user
-  // useEffect(() => {
-  //   if (!currentUser || !results || activeJob) return
+  useMemo(() => {
+    if (!currentUser || userJobAdsLoading || activeJob) return
     
-  //   // user's relevant interests from EAG, such as "software engineering"
-  //   const userEAGInterests = union(currentUser.experiencedIn, currentUser.interestedIn)
-  //   // the topics that the user has displayed on their profile
-  //   const userTags = currentUser.profileTagIds ?? []
-  //   const userJobAds = results[0]?.jobAds ?? {}
+    // user's relevant interests from EAG, such as "software engineering"
+    // TODO: add this back in once we have the data
+    // const userEAGInterests = union(currentUser.experiencedIn, currentUser.interestedIn)
+    const ads = userJobAds ?? []
     
-  //   for (let jobName in JOB_AD_DATA) {
-  //     // skip any jobs where the deadline to apply has passed
-  //     const deadline = JOB_AD_DATA[jobName].deadline
-  //     if (deadline && moment().isAfter(deadline, 'day')) {
-  //       continue
-  //     }
-      
-  //     const eagOccupations = JOB_AD_DATA[jobName].eagOccupations
-  //     const interestedIn = JOB_AD_DATA[jobName].interestedIn
-  //     const occupationTag = JOB_AD_DATA[jobName].tagId
-  //     const jobAdState = userJobAds[jobName]?.state
-  //     // check if the ad fits the user's interests
-  //     const userIsMatch = intersection(userEAGInterests, eagOccupations).length ||
-  //       intersection(currentUser.interestedIn, interestedIn).length ||
-  //       (occupationTag && userTags.includes(occupationTag))
-  //     // make sure the user hasn't already clicked "interested" or "uninterested" for this ad
-  //     const shouldShowAd = !jobAdState || ['seen', 'expanded'].includes(jobAdState)
+    for (let jobName in JOB_AD_DATA) {
+      // skip any jobs where the deadline to apply has passed
+      const deadline = JOB_AD_DATA[jobName].deadline
+      if (deadline && moment().isAfter(deadline, 'day')) {
+        continue
+      }
+
+      const tagsReadIds = JOB_AD_DATA[jobName].tagsReadIds
+      const jobAdState = ads.find(ad => ad.jobName === jobName)?.adState
+      // check if the ad fits the user's interests -
+      // currently based on whether they have subscribed to all the topics relevant to the job ad
+      const userTagSubs = currentUser.frontpageFilterSettings?.tags?.filter(
+        (setting: FilterTag) => filterModeIsSubscribed(setting.filterMode)
+      )?.map((setting: FilterTag) => setting.tagId)
+      const userIsMatch = tagsReadIds && !difference(tagsReadIds, userTagSubs).length
+      // TODO: We probably want to enable this, but not in the initial release, so commenting out for now.
+      // const userIsMatch = coreTagReads &&
+      //   tagsReadIds?.every(
+      //     tagId => coreTagReads.some(tag => tag.tagId === tagId && tag.userReadCount >= 12)
+      //   )
```

Comment:
`coreTagReads`, and probably location-based targeting, will be in a follow up PR